src/extensions/idle.rs

```diff
@@ -13,8 +15,31 @@ use std::time::Duration;
 ///
 /// The handle blocks using the [`IDLE` command](https://tools.ietf.org/html/rfc2177#section-3)
 /// specificed in [RFC 2177](https://tools.ietf.org/html/rfc2177) until the underlying server state
-/// changes in some way. While idling does inform the client what changes happened on the server,
-/// this implementation will currently just block until _anything_ changes, and then notify the
+/// changes in some way.
+///
+/// Each of the `wait` functions takes a callback function which receives any responses
+/// that arrive on the channel while IDLE. The callback function implements whatever
+/// logic is needed to handle the IDLE response, and then returns a [`CallbackAction`]
+/// to `Continue` or `Stop` listening on the channel.
+/// For users that want the IDLE to exit on any change (the behavior proior to version 3.0),
+/// a convenience callback function `callback_stop` is provided.
+///
+/// ```no_run
+/// # use native_tls::TlsConnector;
+/// use imap::extensions::idle;
+/// let ssl_conn = TlsConnector::builder().build().unwrap();
+/// let client = imap::connect(("example.com", 993), "example.com", &ssl_conn)
```

Comment:
Seeing this `TlsConnetor` business all over, I wonder if we should switch (in a separate PR) to a builder pattern with convenience constructors for common TLS implementations. See what we did [in `fantoccini`](https://github.com/jonhoo/fantoccini/pull/115/files), and in particular:
https://github.com/jonhoo/fantoccini/blob/2d1e24acc337e1f0c36e543eb52a20538f1dc18a/src/lib.rs#L180-L196

Reply:
Yes, this would be a nice improvement! :)

Reply:
I added a ticket for this so we do not forget: #190